### ISSUE TITLE
Version 2.10 not published, and app not available in Google Play Store for new users

### ISSUE THREAD

**Issue by chrisirwin** [tasomaniac/OpenLinkWith#99]
I noticed 2.10 has been tagged (and listed in github releases page) >1 year ago. However, it does not actually appear to be available via any discoverable method?

* Google Play Store is offering me version 2.8 (Checked via Pixel 6a, Android 13)
* F-droid has 2.9
* IzzyOnDroid also has 2.9
* Github Releases page has source code, but no built .apk file (previous releases have built .apk files)

Barring setting up my own android build environment, is there a pre-built release available anywhere?

This also causes the app to be undiscoverable and uninstallable in the Google Play Store (for new users) because the targetSdkVersion is >2 years old. This is fixed already in 2.10 (commit 81575790b5bba4a953b42c1f042d96b3fc86f979 bumps targetSdkVersion from 29 to 32). But again, this doesn't appear to be published in the Google Play Store.

Reference on the targetSdkVersion: https://support.google.com/googleplay/android-developer/answer/11926878?hl=en

**Comment by Dhyfer1** [comment]
Hi @chrisirwin 

Yes, you are right. I don't understand why the dev hasn't updated the app to the latest version on Play Store and FDroid. Here we have the source code for the latest version, but I don't know how to compile an app with the source code and I would like to learn how to do it, because here in Github there is another app called [LADB](https://github.com/tytydraco/LADB) that is paid in Play Store, and free if you compile the source code. That's why I would like to learn how to compile based on the source code. If you know how to do it, can you please teach me?

>Barring setting up my own android build environment, is there a pre-built release available anywhere?

I have been searching the internet for the latest version 2.10 and someone on the Russian [4PDA](https://4pda.to/forum/index.php?showtopic=716005#:~:text=app%2Dfloss%2Drelease.apk%20(%202.45%20%D0%9C%D0%91%20)) forum has compiled the app using the source code found in this repository, and uploaded the apk file. To download the apk you have to create an account on the forum, or you can download the apk [here](https://1drv.ms/u/s!AmXb7Ei4MzJflYEjlgDWTcPCdcCqnA?e=RcxgXj).

**Comment by yochananmarqos** [comment]
@Dhyfer1 Both links seem to be dead. Does anyone have a copy of 2.10?

**Comment by Dhyfer1** [comment]
@yochananmarqos I downloaded the file again from the 4pda forum. You can download the file from this [link](https://we.tl/t-Ah91nq0KnS)

**Comment by yochananmarqos** [comment]
@Dhyfer1 Thanks!

**Comment by Dhyfer1** [comment]
> @Dhyfer1 Thanks!

You're welcome bro 👍

**Comment by buttercookie42** [comment]
I've created an unoffical v2.11, which includes the current `master` of this repository plus pull request #97:
https://github.com/buttercookie42/OpenLinkWith/releases/tag/2.11

**Comment by BRBsoup** [comment]
> I've created an unoffical v2.11, which includes the current `master` of this repository plus pull request #97: https://github.com/buttercookie42/OpenLinkWith/releases/tag/2.11

Amazing thanks dude

**Comment by davidgro** [comment]
Thank you @buttercookie42, that should help a lot.

Although I personally found a different app that does everything this one does and a lot more (although it can be more complex to use) - I don't know the rules here on "advertising" (though I have no stake in it) so I won't mention the name unless someone asks. But I did want to bring up that besides the very welcome unofficial build, alternatives also exist. The one I'm using now is also Open Source like OpenLinkWith is.

**Comment by Dhyfer1** [comment]
@davidgro I would like to know what app you are talking about, please

**Comment by BRBsoup** [comment]
@Dhyfer1 Probably BetterOpenWith

**Comment by davidgro** [comment]
@Dhyfer1 URLCheck

Kinda ugly but very functional.

**Comment by SamLeatherdale** [comment]
I know this thread is a bit old, but I just wanted to let you know that I forked @buttercookie42's version of the app, updated it to the latest Android SDK, and relaunched it!

[Open Link With: Redux](https://play.google.com/store/apps/details?id=com.samleatherdale.openwith) is now live on Google Play, and the source is available at [SamLeatherdale/OpenLinkWith](https://github.com/SamLeatherdale/OpenLinkWith) 🚀

**Comment by TPS** [comment]
@SamLeatherdale Would you allow "Issues" for your fork?  Also, think of including in main F-Droid or @IzzySoft's repo?

**Comment by SamLeatherdale** [comment]
Sure, feel free to report issues, I am looking at tackling some of the ones in this repo, starting with disabling the link in shortening by default. However I can't make any guarantees about how quickly that will happen.

I could also publish to F-Droid or IzzyDroid, although I hope it wouldn't lead to confusion as the old version of the app is still there.

So far there have been no functionality changes, asides from upgrading to the latest SDK, but that is prone to change over time. 

I'm not planning on making major overhauls to the app, there are other ones out there like [LinkSheet](https://github.com/LinkSheet/LinkSheet) that have a ton of features that I'm not really trying to compete with, just wanted to make this one available on the Play Store again.

**Comment by SamLeatherdale** [comment]
Oops, TIL it wasn't on by default. Enabled now!

**Comment by tasomaniac** [comment]
Great that you guys carry the work. I'm glad you somehow got the app approved despite all the difficulties Google brought recently

**Comment by IzzySoft** [comment]
> Would you allow "Issues" for your fork? Also, think of including in main F-Droid or @IzzySoft's repo?

As outlined in the other issue: happy to, but we'd need the APKs attached to their corresponding releases.

>  I hope it wouldn't lead to confusion as the old version of the app is still there.

We can take care for that. Your fork uses its own, dedicated appId, right? So we'd put up a note on the original app pointing to the switch, keep it that way for about a month or so, then remove the original one (hope that's OK with you, too, @tasomaniac).

**Comment by tasomaniac** [comment]
Yes for sure please go ahead. That version is pretty old anyways

**Comment by IzzySoft** [comment]
Thanks! Looking forward to the ping for the APK by @SamLeatherdale then.

**Comment by SamLeatherdale** [comment]
@IzzySoft Should I continue to use a separate App ID for the FOSS version? 
For example @tasomaniac had `com.tasomaniac.openwith` for the Play distribution, and `com.tasomaniac.openwith.floss` for the F-Droid distribution? I assume so, so one doesn't overwrite the other, as they are different versions (without Firebase/GMS included)

**Comment by TPS** [comment]
@SamLeatherdale I vote for keeping `com.samleatherdale.openwith` on GPlay & continue the previous/established pattern w/ `com.samleatherdale.openwith.floss` for a (hopefully!) reproducible-build for @IzzySoft's (& perhaps even eventually the main) F-Droid repo.

**Comment by tasomaniac** [comment]
Yeah, I chose to have a different application id because they're essentially different apps because one doesn't have play services. And that also makes Gradle set up a bit easier.

**Comment by SamLeatherdale** [comment]
OK, I have done the same and made it available under [com.samleatherdale.openwith.floss](https://github.com/SamLeatherdale/OpenLinkWith/releases/tag/v3.2.0)

@tasomaniac I did have a bit of a nightmare with setting up Gradle to exclude Play Services, I finally got something working here: https://github.com/SamLeatherdale/OpenLinkWith/commit/a61e088be49f8242fcb5578589ef3d6a8de3ee89

Your original version was so simple, it just had:
```gradle
ext.isPlay = project.hasProperty('play')
```

Where does that project property come from?

**Comment by IzzySoft** [comment]
@SamLeatherdale @tasomaniac to continue the same appIds, there must be an "official transfer of ownership". From the context, this seems what is happening here – but could you, @tasomaniac, please confirm explicitly? And would it be possible to also transfer the signing key and its credentials (on a secure channel of course, not publicly)? Of course only needed if the appIds should be kept. It seems that @SamLeatherdale already has decided to switch, in which case you can ignore this comment 😉

**Comment by SamLeatherdale** [comment]
I thought we are using a different app ID, which would be `com.samleatherdale.openwith.floss`, in which case using a different signing certificate would be expected?

I don't believe @tasomaniac wishes to use the same App ID or signing key

**Comment by tasomaniac** [comment]
Yeah, I don't think so. Just delete my old app completely. So there is no need to transfer the keys or anything. Mine should be deleted just to avoid confusion for users (as discussed earlier in this thread)

About the property, it is something that I just pass along in the cli when building. I made the floss one the default so that play services and such cannot be included accidentally. When you add '-Pplay' while building, it builds for play.

**Comment by IzzySoft** [comment]
Thanks, already implemented. The new app will go live today (and is reproducible 🥳), the original one has a hint added to its description and will be removed after a proper overlap (~1 month) to give people the chance to notice the change.
